pkg/segment/query/metricsquery.go

```diff
@@ -221,6 +227,34 @@ func GetAllMetricNamesOverTheTimeRange(timeRange *dtu.MetricsTimeRange, orgid ui
 	return result, gErr
 }
 
+func applyTagValuesSearchOnlyOnSegments(mQuery *structs.MetricsQuery, allSearchReqests map[string][]*structs.MetricsSearchRequest,
+	mRes *mresults.MetricsResult, timeRange *dtu.MetricsTimeRange, qid uint64, querySummary *summary.QuerySummary) {
+
+	tagValuesRes := make(map[string]map[string]struct{})
+
+	for baseDir := range allSearchReqests {
+		attr, err := tagstree.InitAllTagsTreeReader(baseDir)
+		if err != nil {
+			mRes.AddError(err)
+			continue
+		}
+		sTime := time.Now()
+		err = attr.FindTagValuesOnly(mQuery, tagValuesRes)
```

Comment:
we can just pass `mRes.TagValues` here and don't need to create another map `tagValuesRes`